### ISSUE TITLE
--exclude <file> --check <file> doesn't exclude the file

### ISSUE THREAD

**Issue by mroeschke** [psf/black#949]
Operating system: OSX
Python version: 3.6.8
*Black* version: 19.3b0

```
me:~ me$ echo "a = '123'" > test.py
me:~ me$ cat test.py
a = '123'
me:~ me$ black -l 2 --exclude test.py --check test.py
would reformat test.py
All done! 💥 💔 💥
1 file would be reformatted.
```

Understandably the use case is a bit odd, but in this toy example I would expect `test.py` to be excluded from the check and a message that no files needs to be reformatted.

**Comment by JelleZijlstra** [comment]
I agree that it makes more sense to exclude the file in this case. In general it's safer for Black to exclude more files instead of reformatting files that shouldn't be reformatted.

**Comment by zsol** [comment]
Interesting, I actually expect all files given on the command line to be formatted, because I associate include/exclude with source collecting. Don't have a strong opinion though. I'll meditate on it :)

**Comment by JelleZijlstra** [comment]
I could see myself doing something like `black --exclude=some_dir/ $(find . -name '*.py')`, and there I'd want it to obey the exclude rule.

This has also come up in the past in the context of tools like pre-commit, which want to run Black on all changed files in git without having to re-implement Black's exclude file parsing.

**Comment by mroeschke** [comment]
@JelleZijlstra example is exactly the use case where I discovered this behavior; when using Black in a pre-commit hook context.

**Comment by xmnlab** [comment]
is already there any workaround for this problem?

**Comment by xmnlab** [comment]
probably also related to #986

**Comment by steka** [comment]
As I wrote in the now closed issue #1025, it also useful when used in an editor integration (for example [PyCharm/IntelliJ IDEA](https://black.readthedocs.io/en/stable/editor_integration.html#pycharm-intellij-idea)), where the arguments can just be specified as: "--exclude file.py $FilePath$" to ensure that file.py is never reformatted, even if edited and saved.

**Comment by zsol** [comment]
I just realized there's already an issue open for this problem. Let's move the discussion over to #438.
